homeassistant/components/amberelectric/coordinator.py

```diff
@@ -0,0 +1,112 @@
+"""Amber Electric Coordinator."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from amberelectric import ApiException
+from amberelectric.api import amber_api
+from amberelectric.model.actual_interval import ActualInterval
+from amberelectric.model.channel import ChannelType
+from amberelectric.model.current_interval import CurrentInterval
+from amberelectric.model.forecast_interval import ForecastInterval
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import LOGGER
+
+
+def is_current(interval: ActualInterval | CurrentInterval | ForecastInterval) -> bool:
+    """Return true if the supplied interval is a CurrentInterval."""
+    return interval.__class__ == CurrentInterval
+
+
+def is_forecast(interval: ActualInterval | CurrentInterval | ForecastInterval) -> bool:
+    """Return true if the supplied interval is a ForecastInterval."""
+    return interval.__class__ == ForecastInterval
+
+
+def is_general(interval: ActualInterval | CurrentInterval | ForecastInterval) -> bool:
+    """Return true if the supplied interval is on the general channel."""
+    return interval.channel_type == ChannelType.GENERAL
+
+
+def is_controlled_load(
+    interval: ActualInterval | CurrentInterval | ForecastInterval,
+) -> bool:
+    """Return true if the supplied interval is on the controlled load channel."""
+    return interval.channel_type == ChannelType.CONTROLLED_LOAD
+
+
+def is_feed_in(interval: ActualInterval | CurrentInterval | ForecastInterval) -> bool:
+    """Return true if the supplied interval is on the feed in channel."""
+    return interval.channel_type == ChannelType.FEED_IN
+
+
+class AmberUpdateCoordinator(DataUpdateCoordinator):
+    """AmberUpdateCoordinator - In charge of downloading the data for a site, which all the sensors read."""
+
+    def __init__(
+        self, hass: HomeAssistant, api: amber_api.AmberApi, site_id: str
+    ) -> None:
+        """Initialise the data service."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name="amberelectric",
+            update_method=self.async_update_data,
```

Comment:
This is not necessary since you override it. 
```suggestion
```

Reply:
I checked the code in `update_coordinator.py` and if you don't define `update_method` it'll raise an exception. I've renamed the function in the amber coordinator to  make it clearer.